[ONNX] shape type inference fixes for control flow (#59319)

BowenBao opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #60249 [ONNX] Update documentation (#58712)
* **#60248 [ONNX] shape type inference fixes for control flow (#59319)**
* #60247 [ONNX] Extend chunk for dynamic chunk values (#59644)
* #60246 [ONNX] Add linspace symbolic (#58854)
* #60245 [ONNX] Fix sum export with attribute keepdims (#59316)
* #60244 [ONNX] Fix shape inference for large model (#59320)
* #60243 [ONNX] Handle onnx::Size in ComputeConstant folding (#59122)

* ~~Allow shape inference to skip for blocks by checking unsupported cases recursively. Currently onnx::Identity would trigger a shape inference failure.~~ Fixed in onnx submodule 1.9.
* Remove previous special post process for if op, since that was for constant folding, and now it is handled elsewhere. Update new post process for control flow nodes to copy assign node shape from subblock output shape correctly.

Co-authored-by: BowenBao <bowbao@microsoft.com>

Differential Revision: [D29494913](https://our.internmc.facebook.com/intern/diff/D29494913)

BowenBao committed:
[ONNX] shape type inference fixes for control flow (#59319)

* ~~Allow shape inference to skip for blocks by checking unsupported cases recursively. Currently onnx::Identity would trigger a shape inference failure.~~ Fixed in onnx submodule 1.9.
* Remove previous special post process for if op, since that was for constant folding, and now it is handled elsewhere. Update new post process for control flow nodes to copy assign node shape from subblock output shape correctly.

Co-authored-by: BowenBao <bowbao@microsoft.com>

[ghstack-poisoned]

BowenBao committed:
Update on "[ONNX] shape type inference fixes for control flow (#59319)"


* ~~Allow shape inference to skip for blocks by checking unsupported cases recursively. Currently onnx::Identity would trigger a shape inference failure.~~ Fixed in onnx submodule 1.9.
* Remove previous special post process for if op, since that was for constant folding, and now it is handled elsewhere. Update new post process for control flow nodes to copy assign node shape from subblock output shape correctly.

Co-authored-by: BowenBao <bowbao@microsoft.com>

[ghstack-poisoned]

BowenBao committed:
Update on "[ONNX] shape type inference fixes for control flow (#59319)"


* ~~Allow shape inference to skip for blocks by checking unsupported cases recursively. Currently onnx::Identity would trigger a shape inference failure.~~ Fixed in onnx submodule 1.9.
* Remove previous special post process for if op, since that was for constant folding, and now it is handled elsewhere. Update new post process for control flow nodes to copy assign node shape from subblock output shape correctly.

Co-authored-by: BowenBao <bowbao@microsoft.com>

[ghstack-poisoned]

BowenBao committed:
Update on "[ONNX] shape type inference fixes for control flow (#59319)"


* ~~Allow shape inference to skip for blocks by checking unsupported cases recursively. Currently onnx::Identity would trigger a shape inference failure.~~ Fixed in onnx submodule 1.9.
* Remove previous special post process for if op, since that was for constant folding, and now it is handled elsewhere. Update new post process for control flow nodes to copy assign node shape from subblock output shape correctly.

Co-authored-by: BowenBao <bowbao@microsoft.com>

[ghstack-poisoned]

BowenBao committed:
Update on "[ONNX] shape type inference fixes for control flow (#59319)"


* ~~Allow shape inference to skip for blocks by checking unsupported cases recursively. Currently onnx::Identity would trigger a shape inference failure.~~ Fixed in onnx submodule 1.9.
* Remove previous special post process for if op, since that was for constant folding, and now it is handled elsewhere. Update new post process for control flow nodes to copy assign node shape from subblock output shape correctly.

Co-authored-by: BowenBao <bowbao@microsoft.com>

[ghstack-poisoned]

BowenBao committed:
Update on "[ONNX] shape type inference fixes for control flow (#59319)"


* ~~Allow shape inference to skip for blocks by checking unsupported cases recursively. Currently onnx::Identity would trigger a shape inference failure.~~ Fixed in onnx submodule 1.9.
* Remove previous special post process for if op, since that was for constant folding, and now it is handled elsewhere. Update new post process for control flow nodes to copy assign node shape from subblock output shape correctly.

Co-authored-by: BowenBao <bowbao@microsoft.com>

[ghstack-poisoned]

BowenBao committed:
Update on "[ONNX] shape type inference fixes for control flow (#59319)"


* ~~Allow shape inference to skip for blocks by checking unsupported cases recursively. Currently onnx::Identity would trigger a shape inference failure.~~ Fixed in onnx submodule 1.9.
* Remove previous special post process for if op, since that was for constant folding, and now it is handled elsewhere. Update new post process for control flow nodes to copy assign node shape from subblock output shape correctly.

Co-authored-by: BowenBao <bowbao@microsoft.com>

[ghstack-poisoned]

BowenBao committed:
Update on "[ONNX] shape type inference fixes for control flow (#59319)"


* ~~Allow shape inference to skip for blocks by checking unsupported cases recursively. Currently onnx::Identity would trigger a shape inference failure.~~ Fixed in onnx submodule 1.9.
* Remove previous special post process for if op, since that was for constant folding, and now it is handled elsewhere. Update new post process for control flow nodes to copy assign node shape from subblock output shape correctly.

Co-authored-by: BowenBao <bowbao@microsoft.com>

[ghstack-poisoned]

SplitInfinity commented:
@splitinfinity has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29494913).

BowenBao committed:
Update on "[ONNX] shape type inference fixes for control flow (#59319)"


* ~~Allow shape inference to skip for blocks by checking unsupported cases recursively. Currently onnx::Identity would trigger a shape inference failure.~~ Fixed in onnx submodule 1.9.
* Remove previous special post process for if op, since that was for constant folding, and now it is handled elsewhere. Update new post process for control flow nodes to copy assign node shape from subblock output shape correctly.

Co-authored-by: BowenBao <bowbao@microsoft.com>

Differential Revision: [D29494913](https://our.internmc.facebook.com/intern/diff/D29494913)

[ghstack-poisoned]

SplitInfinity commented:
@splitinfinity has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29494913).
